src/User.cpp

```diff
@@ -19,34 +23,58 @@ std::string User::getAllInfo() const
     ss<<std::left;   
     ss<<std::setw(1) << "1." << name_ << "|";
     ss<<std::setw(1) << "2." << nick_ << "|";
-    ss<<std::setw(1) << "3." << group_ << "|";
-    ss<<std::setw(1) << "4." << gitHub_ << "|";
-    ss<<std::setw(1) << "5." << firecode_ << "|";
+    ss<<std::setw(1) << "3." << mail_ << "|";
+    ss<<std::setw(1) << "4." << group_ << "|";
+    ss<<std::setw(1) << "5." << gitHub_ << "|";
+    ss<<std::setw(1) << "6." << firecode_ << "|";
     ss<<std::endl;
     return ss.str();
 }
+
 void User::setName(std::string name)
 {
     name_ = name;
 }
+
 void User::setNick(std::string nick)
 {
     nick_ = nick;
 }
+
 void User::setGroup(std::string group)
 {
     group_ = group;
 }
+
 void User::setGitHub(std::string gitHub)
 {
     gitHub_ = gitHub;
 }
+
 void User::setFirecode(std::string firecode)
 {
     firecode_ = firecode;
 }
 
-std::string User::getNick() const 
+void User::setPassword(std::string password)
+{
+    password_=password;
+}
+
+void User::setMail(std::string mail)
+{
+    mail_=mail;
+}
+
+std::string User::getNick()  const
 {
     return nick_;
-}
\ No newline at end of file
+}
+
+std::string User::serialize()
+{
+    std::string output{};
+    output = name_+ ';' +nick_ + ';' + password_ + ';' + mail_ + ';' 
+        + group_ + ';' + gitHub_ + ';' + firecode_ + '\n';
+    return output; 
```

Comment:
looks quite similar to getAllInfo(). Maybe one of them is redundant?
ostringstream can be more efficient than appending with + operator.